homeassistant/components/deconz/config_flow.py

```diff
@@ -52,20 +51,29 @@ def __init__(self):
                 if bridge[CONF_HOST] == user_input[CONF_HOST]:
                     self.deconz_config = bridge
                     return await self.async_step_link()
+
             self.deconz_config = user_input
             return await self.async_step_link()
 
         session = aiohttp_client.async_get_clientsession(self.hass)
-        self.bridges = await async_discovery(session)
+
+        try:
+            with async_timeout.timeout(10):
+                self.bridges = await async_discovery(session)
+
+        except asyncio.TimeoutError:
+            self.bridges = []
 
         if len(self.bridges) == 1:
             self.deconz_config = self.bridges[0]
             return await self.async_step_link()
 
         if len(self.bridges) > 1:
             hosts = []
+
             for bridge in self.bridges:
                 hosts.append(bridge[CONF_HOST])
+
             return self.async_show_form(
                 step_id='init',
```

Comment:
This is legacy. There is a lot of work changing from init to user for translation. If I don't use init in the current working way it will not work in the gui

Reply:
Then it is more clear to keep the init step function.